### PR TITLE
Enhancements for Liftbridge operator

### PR DESCRIPTION
**Pull request by annismckenzie**

### CODE REVIEW

**Comment by annismckenzie** [comment]
I dumped every change I needed for the Liftbridge operator to work correctly in here – let's talk about the pieces. I can move the health check probe thing to another MR or we can go with another solution entirely.

File: server/health/health.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package health
+
+import (
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/health"
+	"google.golang.org/grpc/health/grpc_health_v1"
+)
+
+const (
+	ServiceName = "proto.API" // taken from compiled protobuf file api.go.pb (line 793)
```
Comment (tylertreat) [inline server/health/health.go]: Does this need to be exported if it's not used outside the package?

File: server/health/health.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package health
```
Comment (tylertreat) [inline server/health/health.go]: Do we think this warrants its own package?

File: server/health/health.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	"google.golang.org/grpc/health/grpc_health_v1"
+)
+
+const (
+	ServiceName = "proto.API" // taken from compiled protobuf file api.go.pb (line 793)
+)
+
+var server = health.NewServer()
+
+func Register(srv *grpc.Server) {
```
Comment (tylertreat) [inline server/health/health.go]: Please add docstrings to any exported APIs.

File: server/health/health.go (hunk 1 above)
Comment (annismckenzie) [reply server/health/health.go]: Nope, it doesn't – I unexported it.

File: server/health/health.go (hunk 2 above)
Comment (annismckenzie) [reply server/health/health.go]: I tried putting it into the server package but then the methods didn't make any sense anymore and I had to somehow add `health` somewhere in the name – it really wasn't pretty. That's why I went with the extra package.

File: server/health/health.go (hunk 3 above)
Comment (annismckenzie) [reply server/health/health.go]: My bad – added them.

**Comment by annismckenzie** [comment]
May I ask whether you could add me as a collaborator on this project so that the tests can run successfully? 🙈 Or maybe there's a way to configure CircleCI to allow PRs to successfully run even when they need secrets from the main repository?

File: server/health/health.go (hunk 2 above)
Comment (tylertreat) [reply server/health/health.go]: Makes sense.

**Comment by tylertreat** [comment]
@annismckenzie Odd, so it seems CircleCI is not setting the environment variable for your pull requests? I would expect them to be set on all builds. Will take a look...

**Comment by tylertreat** [comment]
I think I've fixed the CI issue.

**Comment by annismckenzie** [comment]
> I think I've fixed the CI issue.

Yup, it's working now.

**Comment by annismckenzie** [comment]
I'll add a test so that the test coverage doesn't decrease. 🙈

**Comment by annismckenzie** [comment]
Tests are now failing because of the rename of the `liftbridge-grpc` module to `liftbridge-api`. I already renamed all the import paths but it still complains with:
```
github.com/liftbridge-io/liftbridge-api: module github.com/liftbridge-io/liftbridge-api@latest (v0.0.0-20190910222614-5694b15f251d) found, but does not contain package github.com/liftbridge-io/liftbridge-api
```

I mean, yes, that project doesn't have a go.mod file…

**Comment by annismckenzie** [comment]
> Tests are now failing because of the rename of the `liftbridge-grpc` module to `liftbridge-api`. I already renamed all the import paths but it still complains with:
> 
> ```
> github.com/liftbridge-io/liftbridge-api: module github.com/liftbridge-io/liftbridge-api@latest (v0.0.0-20190910222614-5694b15f251d) found, but does not contain package github.com/liftbridge-io/liftbridge-api
> ```
> 
> I mean, yes, that project doesn't have a go.mod file…

You'll have to fix all of the import paths in all of the projects until the tests will succeed again.

**Comment by tylertreat** [comment]
Oops, will get that fixed.

**Comment by tylertreat** [comment]
@annismckenzie This should be fixed now. Apologies for the breaking change!

**Comment by annismckenzie** [comment]
Yay, tests are passing again! 🎉

**Comment by annismckenzie** [comment]
YES! 🎉

**Pull request merged**